### DISCUSSION TITLE
How can I nest x-for loops?

### DISCUSSION THREAD

**Discussion by Babelfisch** [alpinejs/alpine#3380] [1. Help] [answered]
I have data structure like this:

```
data: {
  '2023-01-01': [
    { 'title': 'Title 1', 'score': '1' },
    { 'title': 'Title 2', 'score': '2' },
    { 'title': 'Title 3', 'score': '3' },
  ],
  '2023-01-02': [
    { 'title': 'Title 4', 'score': '4' },
    { 'title': 'Title 5', 'score': '5' },
  ],
}
```

This data should be presented in a table like this:

<img width="416" alt="Bildschirm­foto 2023-01-17 um 12 55 10" src="https://user-images.githubusercontent.com/1218405/212892699-82b7c77c-646a-4b82-8005-8598606f8f79.png">

But I have no idea how I can do that with AlpineJS. I tried this code but it doesn't work:

```html
<tbody>
  <template x-for="(scores, date) in data" :key="date">
	<tr>
	  <td colspan="2"><b x-text="date"></b></td>
	</tr>
	<template x-for="(row, index) in scores" :key="index">
	  <tr>
		<td x-text="row.title"></td>
		<td x-text="row.score"></td>
	  </tr>
	</template>
  </template>
</tbody>
```

The inner loop is not visible – I guess because `x-for` has only one child element.

<img width="415" alt="Bildschirm­foto 2023-01-17 um 13 00 08" src="https://user-images.githubusercontent.com/1218405/212893617-51029802-1e37-4f78-8480-58de6d946242.png">

The complete code example is here: https://codesandbox.io/s/keen-browser-gxr9sm

Any idea, how can I realize this?

**Comment by ekwoka** [comment]
Templates can only have one child element. If you wrap the inner content of the x-for here with a div with `display: content` it will work as you expected.

**Reply by Babelfisch** [reply]
It’s a table and a table can’t contain a `<div>` between `<tbody>…</tbody>`.

**Reply by ekwoka** [reply]
right

**Comment by SimoTod** [comment, marked as answer]
Wrap each subsection into a tbody tag. Having multiple tbody tags inside a table is semantically correct
```html
  <template x-for="(scores, date) in data" :key="date">
    <tbody>
	<tr>
	  <td colspan="2"><b x-text="date"></b></td>
	</tr>
	<template x-for="(row, index) in scores" :key="index">
	  <tr>
		<td x-text="row.title"></td>
		<td x-text="row.score"></td>
	  </tr>
	</template>
    </tbody>
  </template>
  ```

**Reply by SimoTod** [reply]
See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tbody#multiple_bodies

**Reply by Babelfisch** [reply]
Thank, good workaround!

**Reply by ekwoka** [reply]
You win this one @SimoTod but this isn't the last you've seen of me!!!!
